### DISCUSSION TITLE
C# Codewars task (Execution Timed Out)

### DISCUSSION THREAD

**Discussion by LunaTiy** [codewars/codewars.com#2675] [General]
### Playing with digits (task):
Some numbers have funny properties. For example:
89 --> 8¹ + 9² = 89 * 1
695 --> 6² + 9³ + 5⁴= 1390 = 695 * 2
46288 --> 4³ + 6⁴+ 2⁵ + 8⁶ + 8⁷ = 2360688 = 46288 * 51

Given a positive integer n written as abcd... (a, b, c, d... being digits) and a positive integer p

we want to find a positive integer k, if it exists, such that the sum of the digits of n taken to the successive powers of p is equal to k * n.
In other words:

Is there an integer k such as : (a ^ p + b ^ (p+1) + c ^(p+2) + d ^ (p+3) + ...) = n * k

If it is the case we will return k, if not return -1.

Note: n and p will always be given as strictly positive integers.

digPow(89, 1) should return 1 since 8¹ + 9² = 89 = 89 * 1
digPow(92, 1) should return -1 since there is no k such as 9¹ + 2² equals 92 * k
digPow(695, 2) should return 2 since 6² + 9³ + 5⁴= 1390 = 695 * 2
digPow(46288, 3) should return 51 since 4³ + 6⁴+ 2⁵ + 8⁶ + 8⁷ = 2360688 = 46288 * 51

My solition passes all tests, but after attempt, it displays a message about the program running time exceeded:
![image](https://user-images.githubusercontent.com/36387585/164991344-c519c759-15dd-4eef-b986-7585a12b7493.png)

#### My repository: https://github.com/LunaTiy/CodewarsTasks/tree/main/PlayingWithDigits
How can I improve the code?

Thanks!

**Comment by Blind4Basics** [comment]
Hi,

Some useful info in that wall of text... ;o
https://docs.codewars.com/training/troubleshooting

Cheers

**Reply by LunaTiy** [reply]
Thanks, I solved this problem by rewriting one method. I don't know why I didn't think of this sooner.
If interesed, I updated the code
